### DISCUSSION TITLE
[auth][error] JWTSessionError in nextJS14 with authjs and prisma + mongodb

### DISCUSSION THREAD

**Discussion by SSazzadur** [nextauthjs/next-auth#10948] [Help] [answered]
### I am getting an error in my project

[auth][error] JWTSessionError: Read more at https://errors.authjs.dev#jwtsessionerror
[auth][cause]: TypeError: Cannot read properties of undefined (reading 'name')
    at Object.jwt (webpack-internal:///(middleware)/./src/lib/auth.config.ts:39:32)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Module.session (webpack-internal:///(middleware)/./node_modules/.pnpm/@auth+core@0.31.0/node_modules/@auth/core/lib/actions/session.js:27:27)
    at async AuthInternal (webpack-internal:///(middleware)/./node_modules/.pnpm/@auth+core@0.31.0/node_modules/@auth/core/lib/index.js:47:24)
    at async Auth (webpack-internal:///(middleware)/./node_modules/.pnpm/@auth+core@0.31.0/node_modules/@auth/core/index.js:126:34)
    at async handleAuth (webpack-internal:///(middleware)/./node_modules/.pnpm/next-auth@5.0.0-beta.18_next@14.2.3_react@18.3.1/node_modules/next-auth/lib/index.js:134:29)
    at async adapter (webpack-internal:///(middleware)/./node_modules/.pnpm/next@14.2.3_react-dom@18.3.1_react@18.3.1/node_modules/next/dist/esm/server/web/adapter.js:179:16)
    at async /Users/.../node_modules/.pnpm/next@14.2.3_react-dom@18.3.1_react@18.3.1/node_modules/next/dist/server/web/sandbox/sandbox.js:110:22
    at async runWithTaggedErrors (/Users/.../node_modules/.pnpm/next@14.2.3_react-dom@18.3.1_react@18.3.1/node_modules/next/dist/server/web/sandbox/sandbox.js:107:9)
    at async DevServer.runMiddleware (/Users/.../node_modules/.pnpm/next@14.2.3_react-dom@18.3.1_react@18.3.1/node_modules/next/dist/server/next-server.js:1062:24)
    
### db.ts
```typescript

import { PrismaClient } from "@prisma/client/edge";

const prismaClientSingleton = () => {
	return new PrismaClient();
};

declare global {
	var prisma: undefined | ReturnType<typeof prismaClientSingleton>;
}

const db = globalThis.prisma ?? prismaClientSingleton();

export default db;

if (process.env.NODE_ENV !== "production") globalThis.prisma = db;


``` 

    
### auth.config.ts
```typescript

import { createUser, fetchUsersWithEmail } from "@/app/_actions/users";
import { NextAuthConfig } from "next-auth";
import Google from "next-auth/providers/google";

import { PrismaAdapter } from "@auth/prisma-adapter";
import db from "./db";

export const authConfig: NextAuthConfig = {
	adapter: PrismaAdapter(db),
	session: { strategy: "jwt" },
	providers: [Google],
	pages: {
		signIn: "/login",
	},
	callbacks: {
		async session({ session, token }) {
			if (token) {
				session.user.name = token.name!;
				session.user.email = token.email!;
				session.user.image = token.picture!;
			}

			return session;
		},

		async jwt({ token, user }) {
			const dbUser = await fetchUsersWithEmail(user?.email || "");

			if (!dbUser) {
				const res = await createUser({ name: user.name!, email: user.email!, image: user.image! });
				token.id = res.id;
				return token;
			}

			console.log("dbUser", dbUser);

			return {
				id: dbUser.id,
				name: dbUser.name,
				email: dbUser.email,
				image: dbUser.image,
			};
		},

		redirect() {
			return "/";
		},
	},
};
```

### auth.ts

```typescript
import NextAuth from "next-auth";
import { authConfig } from "./auth.config";

export const { handlers, signIn, signOut, auth } = NextAuth(authConfig);
```

### middleware.ts

```typescript
import { auth } from "@/lib/auth";
import { NextResponse } from "next/server";

export default auth(req => {
	const pathname = req.nextUrl.pathname;

	// manage route protection
	const isAuth = req.auth;
	const isLoginPage = pathname.startsWith("/login");

	const sensitiveRoutes = ["/"];
	const isAccessingSensitiveRoute = sensitiveRoutes.some(route => pathname.startsWith(route));

	if (isLoginPage) {
		if (isAuth) {
			return NextResponse.redirect(new URL("/", req.url));
		}

		return NextResponse.next();
	}

	if (!isAuth && isAccessingSensitiveRoute) {
		return NextResponse.redirect(new URL("/login", req.url));
	}

	if (pathname === "/") {
		return NextResponse.redirect(new URL("/", req.url));
	}
});

export const config = {
	matcher: ["/", "/login"],
};
```

**Comment by drxc00** [comment]
did you manage to find a solution for this? Im also experiencing the same issue when initializing a middleware. Im using mongoose for object modeling

**Reply by SSazzadur** [reply, marked as answer]
Yes, I found a solution for implementing Google authentication in a Next.js 14 application using NextAuth.js 5 (beta), Prisma ORM, and MongoDB. Check out [this article on medium](https://medium.com/@sazzadur/implementing-google-authentication-in-a-nextjs-14-application-with-authjs-5-mongodb-and-prisma-bbfcb38b3eea)

**Comment by walinabizada** [comment]
![Annotation 2024-12-18 220816](https://github.com/user-attachments/assets/31cdf8e4-397a-45b4-89f6-fca870d34665)

just clear the cookies, when you run the project on **browser -> inspect element Application -> Cookies** right click **clear**, the issue solved. that is all, I have the same issue.
